### DISCUSSION TITLE
Support for custom header and footer using URL into PDF

### DISCUSSION THREAD

**Discussion by kovacevjosip** [gotenberg/gotenberg#1024] [Q&A]
Hi,
by documentation, it is only possible to send `header` and `footer` as file.
When using [URL into PDF ](https://gotenberg.dev/docs/routes#url-into-pdf-route) it is a bit unusual to send header and footer as files, especially when some dynamic content is used.

Generally we have mini app that renders templates using URL to PDF format, where header and footer are detached. For example:
- https://.../template
- https://.../template/header
- https:://.../template/footer

I know that I can fetch header and footer first and then send them as files, but is there any other (more simple) solution?
Note that I need support for `pageNumber` and `totalPages`.

Thanks for your awesome support!

**Comment by gulien** [comment]
Hello @kovacevjosip 👋

Maybe the `downloadFrom` form field is what you’re looking for? It may require some adjustment on you side  though.

In any case, this is a Chromium feature and Chromium is waiting for HTML, not a pre-rendered URL ^^

**Comment by kovacevjosip** [comment]
Thanks for fast response. `downloadFrom` is interesting approach, potentially it could work.
I ended up using [pagedjs](https://pagedjs.org/) in combination with `gotenberg`. It is very flexible and nicely documented.
